src/unix.rs

```diff
@@ -154,40 +122,25 @@ extern "C" fn handler(signum: c_int, info: *mut libc::siginfo_t, ptr: *mut libc:
 /// This will register the signal handler if it hasn't already been registered,
 /// returning any error along the way if that fails.
 fn signal_enable(signal: c_int) -> io::Result<()> {
-    let siginfo = match globals().signals.get(signal as usize) {
+    let globals = globals();
+    let siginfo = match globals.signals.get(signal as usize) {
         Some(slot) => slot,
         None => return Err(io::Error::new(io::ErrorKind::Other, "signal too large")),
     };
-    unsafe {
-        let mut err = None;
-        siginfo.init.call_once(|| {
-            let mut new: libc::sigaction = mem::zeroed();
-            new.sa_sigaction = handler as usize;
-            let flags = libc::SA_RESTART | libc::SA_NOCLDSTOP;;
-            // android defines SIGINFO with a different type than sa_flags,
-            // this ensures that the variables are of the same type regardless of platform
-            #[allow(unused_assignments)]
-            let mut sa_siginfo = flags;
-            sa_siginfo = libc::SA_SIGINFO as _;
-            let flags = flags | sa_siginfo;
-            new.sa_flags = flags as _;
-            if libc::sigaction(signal, &new, &mut *siginfo.prev.get()) != 0 {
-                err = Some(io::Error::last_os_error());
-            } else {
-                *siginfo.initialized.get() = true;
-            }
-        });
-        if let Some(err) = err {
-            return Err(err);
-        }
-        if *siginfo.initialized.get() {
-            Ok(())
-        } else {
-            Err(io::Error::new(
-                io::ErrorKind::Other,
-                "failed to register signal handler",
-            ))
+    let mut registered = Ok(());
+    siginfo.init.call_once(|| {
+        registered = unsafe {
+            signal_hook::register(signal, move || action(siginfo, &globals.sender)).map(|_| ())
```

Comment:
Before doing this registration, can we return an error if `signal` happens to be one of the `signal_hook::FORBIDDEN` signals?

Although the forbidden signals probably won't do what the caller wants, I don't want to start suddenly start panicking without at least bumping semver. 

Once we get to `0.3.0` we can remove this error check and have `signal_hook` panic for us (You can leave a `FIXME(breaking):` note around the check if you'd like)